src/backend/favourites/favourites-section.c

```diff
@@ -0,0 +1,155 @@
+ */
+
+#define _GNU_SOURCE
+
+#include "util.h"
+
+BRISK_BEGIN_PEDANTIC
+#include "favourites-section.h"
+#include <gio/gio.h>
+#include <glib/gi18n.h>
+BRISK_END_PEDANTIC
+
+struct _BriskFavouritesSectionClass {
+        BriskSectionClass parent_class;
+};
+
+struct _BriskFavouritesSection {
+        BriskSection parent;
+        GIcon *icon; /**<Display icon */
+        GSettings *settings;
+};
+
+G_DEFINE_TYPE(BriskFavouritesSection, brisk_favourites_section, BRISK_TYPE_SECTION)
+
+/* Helper for gsettings */
+typedef gchar *gstrv;
+DEF_AUTOFREE(gstrv, g_strfreev)
+
+/**
+ * Basic subclassing
+ */
+static const gchar *brisk_favourites_section_get_id(BriskSection *item);
+static const gchar *brisk_favourites_section_get_name(BriskSection *item);
+static const GIcon *brisk_favourites_section_get_icon(BriskSection *item);
+static const gchar *brisk_favourites_section_get_backend_id(BriskSection *item);
+static gboolean brisk_favourites_section_can_show_item(BriskSection *section, BriskItem *item);
+
+/**
+ * brisk_favourites_section_dispose:
+ *
+ * Clean up a BriskFavouritesSection instance
+ */
+static void brisk_favourites_section_dispose(GObject *obj)
+{
+        BriskFavouritesSection *self = BRISK_FAVOURITES_SECTION(obj);
+
+        g_clear_object(&self->icon);
+
+        G_OBJECT_CLASS(brisk_favourites_section_parent_class)->dispose(obj);
+}
+
+/**
+ * brisk_favourites_section_class_init:
+ *
+ * Handle class initialisation
+ */
+static void brisk_favourites_section_class_init(BriskFavouritesSectionClass *klazz)
+{
+        BriskSectionClass *s_class = BRISK_SECTION_CLASS(klazz);
+        GObjectClass *obj_class = G_OBJECT_CLASS(klazz);
+
+        /* item vtable hookup */
+        s_class->get_id = brisk_favourites_section_get_id;
+        s_class->get_name = brisk_favourites_section_get_name;
+        s_class->get_icon = brisk_favourites_section_get_icon;
+        s_class->get_backend_id = brisk_favourites_section_get_backend_id;
+        s_class->can_show_item = brisk_favourites_section_can_show_item;
+
+        obj_class->dispose = brisk_favourites_section_dispose;
+}
+
+/**
+ * brisk_favourites_section_init:
+ *
+ * Handle construction of the BriskFavouritesSection. Does absolutely nothing
+ * special outside of creating our icon.
+ */
+static void brisk_favourites_section_init(BriskFavouritesSection *self)
+{
+        self->icon = g_themed_icon_new_with_default_fallbacks("emblem-favorite");
+        self->settings = g_settings_new("com.solus-project.brisk-menu");
+}
+
+static gboolean brisk_favourites_section_get_item_is_pinned(BriskSection *section, BriskItem *item)
+{
+        autofree(gstrv) *favourites = NULL;
+
+        BriskFavouritesSection *self = BRISK_FAVOURITES_SECTION(section);
+        favourites = g_settings_get_strv(self->settings, "favourites");
+
+        if (!favourites) {
+                return FALSE;
+        }
+
+        return g_strv_contains((const gchar *const *)favourites, brisk_item_get_id(item));
```

Comment:
I'm worried about the efficiency here tbqh. I can optimize it post-merge though.